addons/test_website_slides_full/tests/test_ui_wslides.py

```diff
@@ -163,9 +131,53 @@ def test_course_certification_employee(self):
                     'is_published': True,
                     'is_preview': False,
                     'description': "It's time to test your knowledge!",
-                    'survey_id': furniture_survey.id,
+                    'survey_id': cls.furniture_survey.id,
                 })
             ]
         })
 
-        self.start_tour('/slides', 'certification_member', login=user_demo.login, timeout=90)
+    @users("portal")
+    def test_course_certification_employee(self):
+        user_portal = self.env.user
+
+        sale_order_data = {
+            'partner_id': user_portal.partner_id.id,
+            'order_line': [
+                (0, 0, {
+                    'name': self.furniture_course_product.name,
+                    'product_id': self.furniture_course_product.id,
+                    'product_uom_qty': 1,
+                    'price_unit': self.furniture_course_product.list_price,
+                })
+            ]}
+
+        # =================
+        # FAILURE ATTEMPTS
+        # =================
+        self.env['sale.order'].sudo().create(sale_order_data).action_confirm()
+        # Member should have access to the course
+        self.assertTrue(self.furniture_course.with_user(user_portal).is_member)
```

Comment:
Should not have to use "with_user"as the test annotation member is already portal.

(Same for others here below).

If it doesn't work otherwise, maybe try invalidating the cache of "self.furniture_course" at the beginning of the method.

Reply:
It's not working (even with invalidate cache), it was always giving `False`...
![image](https://github.com/user-attachments/assets/10b7711f-ea5f-4855-96f4-c01313ce3c49)

Reply:
For test purpose, I checked env user in test function, and I was getting `OdooBot` for some reason....

```py
    @users("portal")
    def test_course_certification_employee(self):
        user_portal = self.env.user
        # self.furniture_course._invalidate_cache()
        test = self.furniture_course.action_redirect_to_certified_members()
        # breakpoint is in action
```
![image](https://github.com/user-attachments/assets/b4cd7304-3ff0-4a1d-bb63-f319ca1cde5f)

I'm not sure why it's happening.. :confused:

Reply:
What if you try at the beginning of the function:

`self.furniture_course = self.env['slide.channel'].browse(self.furniture_course.id)  # use proper environment to test user dependent computes`

If it doesn't work, let's just keep the with_user then, it's fine.

(Don't forget to mention the new timings in the commit message so we can merge this one).

Thanks & cheers!